### ISSUE TITLE
Can't script peer login

### ISSUE THREAD

**Issue by bcbrock** [hyperledger-archives/fabric#617]
Ideally a script like this should work:

```
echo -n NPKYL39uKbkj | OPENCHAIN_PEER_ADDRESS=localhost:30303 ./obc-peer login lukas
```

Instead it crashes (both with and without the -n option to echo)

```
Enter password for user 'lukas': panic: inappropriate ioctl for device

goroutine 1 [running]:
github.com/openblockchain/obc-peer/vendor/github.com/howeyc/gopass.getch(0x0)
    /gsa/ausgsa/home/b/c/bcbrock/obc/code/src/github.com/openblockchain/obc-peer/vendor/github.com/howeyc/gopass/nix.go:13 +0x73
github.com/openblockchain/obc-peer/vendor/github.com/howeyc/gopass.getPasswd(0x0, 0x0, 0x0, 0x0)
    /gsa/ausgsa/home/b/c/bcbrock/obc/code/src/github.com/openblockchain/obc-peer/vendor/github.com/howeyc/gopass/pass.go:18 +0x12e
github.com/openblockchain/obc-peer/vendor/github.com/howeyc/gopass.GetPasswd(0x0, 0x0, 0x0)
    /gsa/ausgsa/home/b/c/bcbrock/obc/code/src/github.com/openblockchain/obc-peer/vendor/github.com/howeyc/gopass/pass.go:37 +0x2d
main.login(0xc8202744c0, 0x1, 0x1, 0x7f8f0dde9ac8, 0xc82024f860)
    /gsa/ausgsa/home/b/c/bcbrock/obc/code/src/github.com/openblockchain/obc-peer/main.go:492 +0x4e1
main.glob.func8(0x135da60, 0xc8202744c0, 0x1, 0x1, 0x0, 0x0)
    /gsa/ausgsa/home/b/c/bcbrock/obc/code/src/github.com/openblockchain/obc-peer/main.go:115 +0x41
github.com/openblockchain/obc-peer/vendor/github.com/spf13/cobra.(*Command).execute(0x135da60, 0xc820274420, 0x1, 0x1, 0x0, 0x0)
    /gsa/ausgsa/home/b/c/bcbrock/obc/code/src/github.com/openblockchain/obc-peer/vendor/github.com/spf13/cobra/command.go:497 +0x63b
github.com/openblockchain/obc-peer/vendor/github.com/spf13/cobra.(*Command).Execute(0x135d360, 0x0, 0x0)
    /gsa/ausgsa/home/b/c/bcbrock/obc/code/src/github.com/openblockchain/obc-peer/vendor/github.com/spf13/cobra/command.go:584 +0x46a
main.main()
    /gsa/ausgsa/home/b/c/bcbrock/obc/code/src/github.com/openblockchain/obc-peer/main.go:256 +0x13f9

goroutine 17 [syscall, locked to thread]:
runtime.goexit()
    /usr/lib/go/src/runtime/asm_amd64.s:1696 +0x1
```

In lieu of or in addition to fixing the crash, it might be nice to have a command-line option for the password to simplify test scripting.

**Comment by bcbrock** [comment]
The panic was fixed in howeyc/gopass 18 days ago, it now returns an `error` in this case. 

From looking at how other programs handle passwords it looks like this error/panic is not unexpected, and has something to do with unix-specific handling of "terminals". I propose to submit a PR that will add a  (-p, --pipe) option to the `obc-peer login` subcommand that will behave like the similar option to the gsa_login command:

```
   -p     Accept password from a pipe: This allows scripts to feed passwords into gsa_login through its standard input.  If this flag is not set gsa_login  will
          only accept passwords from its controlling terminal.
```

I've tried these fixes and know they will work. @binhn - How do you guys update your "vendor" code? Simply re-checkout the latest version and commit any diffs - or something other?

**Comment by srderson** [comment]
We use [Govendor](https://github.com/kardianos/govendor). It's already installed in the Vagrant image. There's an 'Adding or updating a Go packages' section in the [README](https://github.com/openblockchain/obc-peer/blob/master/README.md)

**Comment by binhn** [comment]
@bcbrock Do we still need work on this or can be closed?

**Comment by bcbrock** [comment]
@binhn - This issue is still open. I've been working on other things and have not gotten to it myself.

**Comment by genggjh** [comment]
You can assign it to me, I will try to fix this.

**Comment by markparz** [comment]
@binhn can you assign @genggjh to this one? Thanks Guys!

**Comment by srderson** [comment]
Closed via PR #1090
